src/com/owncloud/android/lib/common/OwnCloudClient.java

```diff
@@ -199,62 +200,85 @@ public int executeMethod(HttpMethodBase method, int readTimeout, int connectionT
-            }
+        int status = super.executeMethod(method);
 
-//	        logCookiesAtRequest(method.getRequestHeaders(), "after");
-//	        logCookiesAtState("after");
-//	        logSetCookiesAtResponse(method.getResponseHeaders());
+        if (mFollowRedirects) {
+            status = followRedirection(method).getLastStatus();
+        }
 
-            return status;
+        //logCookiesAtRequest(method.getRequestHeaders(), "after");
+        //logCookiesAtState("after");
+        //logSetCookiesAtResponse(method.getResponseHeaders());
 
-        } catch (IOException e) {
-            //Log_OC.d(TAG + " #" + mInstanceNumber, "Exception occurred", e);
-            throw e;
+        return status;
+    }
+
+
+    /**
+     * Fix for https://github.com/owncloud/android/issues/1847#issuecomment-267558274
+     *
+     * The problem: default SocketFactory in HTTPClient 3.x for HTTP connections creates a separate thread
+     * to create the socket. When a port out of TCP bounds is passed, an exception is thrown in that
+     * separate thread, and our original thread is not able to catch it. This is not happenning with HTTPS
+     * connections because we had to define our own socket factory,
+     * {@link com.owncloud.android.lib.common.network.AdvancedSslSocketFactory}, and it does not mess with
+     * threads.
+     *
+     * The solution: validate the input (the port number) ourselves before let the work to HTTPClient 3.x.
+     *
+     * @param method HTTP method to run.
+     * @throws IllegalArgumentException If 'method' targets an invalid port in an HTTP URI.
+     * @throws URIException             If the URI to the target server cannot be built.
+     */
+    private void preventCrashDueToInvalidPort(HttpMethod method) throws URIException {
+        int port = method.getURI().getPort();
+        String scheme = method.getURI().getScheme().toLowerCase();
+        if ("http".equals(scheme) && port > 0xFFFF) {
+            // < 0 is not tested because -1 is used when no port number is specified in the URL;
+            // no problem, the network library will convert that in the default HTTP port
+            throw new IllegalArgumentException("Invalid port number " + port);
         }
     }
 
 
-	public RedirectionPath followRedirection(HttpMethod method) throws IOException {
+    public RedirectionPath followRedirection(HttpMethod method) throws IOException {
         int redirectionsCount = 0;
         int status = method.getStatusCode();
         RedirectionPath result = new RedirectionPath(status, MAX_REDIRECTIONS_COUNT);
         while (redirectionsCount < MAX_REDIRECTIONS_COUNT &&
-                (   status == HttpStatus.SC_MOVED_PERMANENTLY || 
-                    status == HttpStatus.SC_MOVED_TEMPORARILY ||
-                    status == HttpStatus.SC_TEMPORARY_REDIRECT)
-                ) {
-            
+            (status == HttpStatus.SC_MOVED_PERMANENTLY ||
```

Comment:
This OR condition is too long. You could create a variable (boolean HttpStatusOK) with this condition before the while bucle starts and after that, add the variable to the while bucle.